internal/common/warnings.go

```diff
@@ -3,12 +3,17 @@ package common
 
 import "fmt"
 
+const maxWarnings = 100 // An arbitrary limit to avoid excessive memory usage, it has no sense to store thousands of errors
+
 type Warnings struct {
 	List    []error
 	Verbose bool
 }
 
 func (w *Warnings) Add(err error) {
+	if len(w.List) >= maxWarnings {
+		return
+	}
 	w.List = append(w.List, err)
 }
```

Comment:
It's a good idea, but then I would prefer it's not a silent behavior.


```suggestion
const maxWarnings = 100 // An arbitrary limit to avoid excessive memory usage, it has no sense to store thousands of errors

 type Warnings struct {
 	List    []error
	TooManyErrors bool
 	Verbose bool
 }
 
 func (w *Warnings) Add(err error) {
	if len(w.List) >= maxWarnings {
		w.TooManyErrors = true
		return
	}
 	w.List = append(w.List, err)
 }
```

Or something like this

```suggestion
const maxWarnings = 100 // An arbitrary limit to avoid excessive memory usage, it has no sense to store thousands of errors

 type Warnings struct {
 	List    []error
	tooManyErrors bool
 	Verbose bool
 }
 
var ErrTooManyErrors = errors.New("too many errors reported")

 func (w *Warnings) Add(err error) {
	if len(w.List) >= maxWarnings {
		if w.tooManyErrors {
			return
		}
		w.tooManyErrors = true
		err = ErrTooManyErrors
	}
 	w.List = append(w.List, err)
 }
```

What do you guys think about this?

Reply:
it's ok to me, lets wait @shirou opinion

Reply:
👍 since `go build` shows the same behavior when there are many build errors.
However, I don’t think we need to define `ErrTooManyErrors` as public, because the behavior doesn’t really need to change depending on whether there are too many errors or not.

You may have already noticed, but I’d also suggest adding a notice in the `Error()` output, for example:

```
if w.tooManyerrors {
    return fmt.Sprintf("too many errors reported.")
}
```